src/solver/types.jl

```diff
@@ -375,123 +375,58 @@ struct AtmosConfig{FT, TD, PA, C}
-Random.seed!(1234)
-parsed_args = CA.AtmosCoveragePerfParsedArgs(;moist="dry")
-config = CA.AtmosConfig(;parsed_args)
-```
+    AtmosCoveragePerfConfig()
+    AtmosCoveragePerfConfig(s)
+Creates a model configuration for many performance tests.
+Creates a config from the following in order of top priority to last:
+1. Configuration from the given config file
+2. Default perf configuration
+3. Target job configuration
+4. Default configuration
 """
-function AtmosCoveragePerfParsedArgs(s = argparse_settings())
-    @info "Using coverage performance parameters + provided CL arguments."
-    parsed_args_defaults = cli_defaults(s)
-    dict = parsed_args_per_job_id(; filter_name = "driver.jl")
-
-    # Start with performance target, and override anything provided in ARGS
-    parsed_args_prescribed = parsed_args_from_ARGS(ARGS)
-
-    target_job = get(parsed_args_prescribed, "target_job", nothing)
-    parsed_args_perf_target = isnothing(target_job) ? Dict() : dict[target_job]
-
-    parsed_args_perf_target["forcing"] = "held_suarez"
-    parsed_args_perf_target["vert_diff"] = true
-    parsed_args_perf_target["surface_setup"] = "DefaultExchangeCoefficients"
-    parsed_args_perf_target["moist"] = "equil"
-    parsed_args_perf_target["rad"] = "allskywithclear"
-    parsed_args_perf_target["precip_model"] = "0M"
-    parsed_args_perf_target["dt"] = "1secs"
-    parsed_args_perf_target["t_end"] = "10secs"
-    parsed_args_perf_target["dt_save_to_sol"] = Inf
-    parsed_args_perf_target["z_elem"] = 25
-    parsed_args_perf_target["h_elem"] = 12
-
-    parsed_args = merge(
-        parsed_args_defaults,
-        parsed_args_perf_target,
-        parsed_args_prescribed,
-    )
-    return parsed_args
-end
-
-AtmosCoveragePerfConfig(s = argparse_settings()) =
-    AtmosConfig(s; parsed_args = AtmosCoveragePerfParsedArgs(s))
-
-function AtmosConfigArgs(
-    s = argparse_settings();
-    args = String[],
-    parsed_args = parse_commandline(args, s),
-    comms_ctx = get_comms_context(parsed_args),
-)
-    @info "Running ClimaAtmos with default argparse settings + $args"
-    return AtmosConfig(s; parsed_args, comms_ctx)
+function AtmosCoveragePerfConfig(s = argparse_settings())
+    parsed_args = parse_commandline(s)
+    config_dict = merge(YAML.load_file.(parsed_args["config_file"])...)
+    target_job_config = if haskey(config_dict, "target_job")
+        config_from_target_job(config_dict["target_job"])
+    else
+        Dict()
+    end
+    perf_defaults =
+        joinpath(pkgdir(CA), ".buildkite", "perf_configs", "perf_default.yml")
+    perf_default_config = YAML.load_file(perf_defaults)
+    config_dict = merge(target_job_config, perf_default_config, config_dict)
+    return AtmosConfig(; s, config_dict = config_dict)
 end
 
-function AtmosConfig(
-    s = argparse_settings();
+function AtmosConfig(;
+    s = argparse_settings(),
     parsed_args = parse_commandline(s),
-    comms_ctx = get_comms_context(parsed_args),
+    config_dict = nothing,
```

Comment:
Just to note, changing the kwarg name is a breaking change (which is fine, just want to note it), I've added the breaking label.